src/axom/inlet/Inlet.cpp

```diff
@@ -0,0 +1,183 @@
+// Copyright (c) 2017-2020, Lawrence Livermore National Security, LLC and
+// other Axom Project Developers. See the top-level COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+/*!
+ *******************************************************************************
+ * \file Inlet.cpp
+ *
+ * \brief This file contains the class implementation of Inlet, the main class
+ *        for the Inlet component.
+ *******************************************************************************
+ */
+
+#include "axom/inlet/Inlet.hpp"
+
+#include "fmt/fmt.hpp"
+#include "axom/slic.hpp"
+
+namespace axom
+{
+namespace inlet
+{
+
+std::shared_ptr<Table> Inlet::addTable(const std::string& name,
+                                       const std::string& description)
+{
+  return std::make_shared<Table>(name, description, m_reader, m_sidreRootGroup);
+}
+
+std::shared_ptr<Field> Inlet::addBool(const std::string& name,
+                                      const std::string& description)
+{
+  return m_globalTable->addBool(name, description);
+}
+
+std::shared_ptr<Field> Inlet::addDouble(const std::string& name,
+                                        const std::string& description)
+{
+  return m_globalTable->addDouble(name, description);
+}
+
+std::shared_ptr<Field> Inlet::addInt(const std::string& name,
+                                     const std::string& description)
+{
+  return m_globalTable->addInt(name, description);
+}
+
+std::shared_ptr<Field> Inlet::addString(const std::string& name,
+                                        const std::string& description)
+{
+  return m_globalTable->addString(name, description);
+}
+
+
+//-------------------------------------------------
+//   Get values out of the datastore
+//-------------------------------------------------
+
+axom::sidre::View* Inlet::baseGet(const std::string& name)
+{
+  SLIC_ASSERT_MSG(m_sidreRootGroup != nullptr, "Inlet's Sidre Datastore Group not set");
+
+  // All data hangs under the group's name
+  if (!m_sidreRootGroup->hasGroup(name))
+  {
+    return nullptr;
+  }
+  axom::sidre::Group* group = m_sidreRootGroup->getGroup(name);
```

Comment:
Would it make sense to delegate these accesses to the `Table` class?

Reply:
I'm guessing this will be done in future updates? along with changes to the `Field` class ?

Reply:
The idea was that the "get" functions would be used later in the program and the "add" functions would be used when you were defining what should/can be in the input deck.  You're right an example would clear this up.